File: python/diabetes_analysis.py
import pandas as pd

# Load dataset
file_path = 'diabetes.csv'
df = pd.read_csv(file_path)

# Display first rows
print(df.head())

# Average glucose by diabetes outcome
avg_glucose = df.groupby('Outcome')['Glucose'].mean()
print('\nAverage Glucose Levels')
print(avg_glucose)

# Average age by diabetes outcome
avg_age = df.groupby('Outcome')['Age'].mean()
print('\nAverage Age')
print(avg_age)

# BMI analysis
avg_bmi = df.groupby('Outcome')['BMI'].mean()
print('\nAverage BMI')
print(avg_bmi)

# Summary statistics
print('\nSummary Statistics')
print(df.describe())

File: r/diabetes_analysis.R
library(readr)
library(dplyr)

# Load dataset
file_path <- 'D:\\Hello World\\diabetes\\diabetes.csv'
df <- read_csv(file_path)

# View first rows
head(df)

# Average glucose by outcome
avg_glucose <- df %>%
  group_by(Outcome) %>%
  summarise(mean_glucose = mean(Glucose, na.rm = TRUE))

print(avg_glucose)

# Average age by outcome
avg_age <- df %>%
  group_by(Outcome) %>%
  summarise(mean_age = mean(Age, na.rm = TRUE))

print(avg_age)

# Average BMI
avg_bmi <- df %>%
  group_by(Outcome) %>%
  summarise(mean_bmi = mean(BMI, na.rm = TRUE))

print(avg_bmi)

